cli/main.py

```diff
@@ -941,12 +943,10 @@ def __deploy_contract(deployer_class: ClassVar,
                 __deployer_init_args.update({dependant: __deployment_agents[dependant]})
 
             if upgradeable:
-                def __collect_secret_hash():
-                    secret = click.prompt("Enter secret hash for {}".format(__contract_name), hide_input=True, confirmation_prompt=True)
-                    secret_hash = hashlib.sha256(secret)
-                    __deployer_init_args.update({'secret_hash': secret_hash})
-                    return secret
-                __collect_secret_hash()
+                secret = click.prompt("Enter deployment secret for {}".format(__contract_name),
+                                      hide_input=True, confirmation_prompt=True)
+                secret_hash = hashlib.sha256(secret)
```

Comment:
What are the security requirements for the deployment secret (if at all)?

Reply:
Yeah - nothing is enforced here, we can pipe this through a validator first.